src/main/resources/native/android/c/attach_adapter.c

```diff
@@ -33,50 +33,26 @@ jclass jKeyboardServiceClass;
 jclass jPositionServiceClass;
 int handlesInitialized = 0;
 
-void registerAttachMethodHandles(JNIEnv* androidEnv) {
-    if (handlesInitialized > 0) {
-        return;
-    }
-    // Util
-    jclass jtmp = (*androidEnv)->FindClass(androidEnv, "com/gluonhq/helloandroid/Util");
+jclass registerClass(JNIEnv* androidEnv, const char* name) {
+    jclass jtmp = (*androidEnv)->FindClass(androidEnv, name);
     jthrowable t = (*androidEnv)->ExceptionOccurred(androidEnv);
     if (t) {
         (*androidEnv)->ExceptionClear(androidEnv);
     }
     if ((t == NULL) && (jtmp != NULL)) {
-        jUtilClass= (jclass)(*androidEnv)->NewGlobalRef(androidEnv, jtmp);
-    }
-
-    // BLE
-    jtmp = (*androidEnv)->FindClass(androidEnv, "com/gluonhq/helloandroid/BleService");
-    t = (*androidEnv)->ExceptionOccurred(androidEnv);
-    if (t) {
-        (*androidEnv)->ExceptionClear(androidEnv);
-    }
-    if ((t == NULL) && (jtmp != NULL)) {
-        jBleServiceClass= (jclass)(*androidEnv)->NewGlobalRef(androidEnv, jtmp);
+        return (jclass)(*androidEnv)->NewGlobalRef(androidEnv, jtmp);
```

Comment:
Add missing return, In case the if test fails?

Reply:
good catch, fixed.